assets/components/src/plugin-toggle/index.js

```diff
@@ -0,0 +1,171 @@
+				} );
+			}
+		);
+	};
+
+	/**
+	 * Render.
+	 */
+	render() {
+		const { plugins } = this.props;
+		const { pluginInfo } = this.state;
+		return this.prepareDataForRender( plugins, pluginInfo ).map( plugin => {
+			const { name, description, href, slug, status, editPath, inFlight } = plugin;
+			const pluginStatus = this.statusForPlugin( plugin );
+			const handoff = ! href && pluginStatus && editPath ? slug : null;
+			return (
+				<ActionCard
+					className={ this.classNameForPlugin( plugin ) }
+					title={ name }
+					description={ description }
+					actionText={ this.actionTextForPlugin( plugin ) }
+					handoff={ handoff }
+					href={ href }
+					toggle
+					toggleChecked={ this.statusForPlugin( plugin ) }
+					toggleOnChange={ value => this.managePlugin( slug, value ) }
+				/>
+			);
+		} );
+	}
+
+	/**
+	 * Prepare plugins data for render. Change all keys to camelCase, merge API-fetched data with prop.
+	 */
+	prepareDataForRender = ( pluginsFromProps, pluginsFromAPI ) =>
+		Object.keys( pluginsFromProps )
+			.map( pluginSlug =>
+				pluginsFromAPI[ pluginSlug ]
+					? Object.keys( pluginsFromAPI[ pluginSlug ] ).reduce(
+							( accumulator, key ) => ( {
+								...accumulator,
+								[ key.charAt( 0 ).toLowerCase() + key.slice( 1 ) ]: pluginsFromAPI[ pluginSlug ][
+									key
+								],
+							} ),
+							{}
+					  )
+					: {}
+			)
+			.map( plugin => Object.assign( plugin, pluginsFromProps[ plugin.slug ] ) );
+
+	/**
+	 * Generate a classname for the ActionCard based on plugin state. Applies 'in-flight' class if an API is underway and 'loading' if the plugin data is not yet available.
+	 */
+	classNameForPlugin = plugin => {
+		const { status, inFlight } = plugin;
+		if ( inFlight ) {
+			return 'in-flight';
+		}
+		if ( ! status ) {
+			return 'loading';
+		}
+	};
+
+	/**
+	 * Generate the ActionCard action text for a plugin.
+	 */
+	actionTextForPlugin = plugin => {
+		const { actionText, editPath, href, inFlight, name } = plugin;
+		// Show spinner when plugin data is unavailable, or when an API call is in flight.
+		if ( 'configure' === inFlight ) {
+			return (
+				<Fragment>
+					{ __( 'Installing...', 'newspack' ) } <Waiting isRight />
+				</Fragment>
+			);
+		}
+		if ( 'uninstall' === inFlight ) {
+			return (
+				<Fragment>
+					{ __( 'Uninstalling...', 'newspack' ) } <Waiting isRight />
+				</Fragment>
+			);
+		}
+		if ( ! name ) {
+			return (
+				<Fragment>
+					{ __( 'Loading...', 'newspack' ) } <Waiting isRight />
+				</Fragment>
+			);
+		}
+		// No action button at all if the plugin isn't installed and active.
+		if ( ! this.statusForPlugin( plugin ) ) {
+			return null;
+		}
+		if ( href || editPath ) {
+			return actionText ? actionText : __( 'Configure', 'newspack' );
+		}
+	};
+
+	/**
+	 * Get installation/activation state for a plugin.
+	 */
+	statusForPlugin = plugin => {
```

Comment:
nitpick: name's a bit misleading - I'd expect `isStatusActive` / `isPluginActive`

Reply:
Good idea. How's this? https://github.com/Automattic/newspack-plugin/pull/406/commits/cc2aa3100cbaf264e929276fef4837fcb2e69cc9